tests/baselines/reference/api/typescript.d.ts

```diff
@@ -5964,6 +5964,12 @@ declare namespace ts {
         /** Not true for all global completions. This will be true if the enclosing scope matches a few syntax kinds. See `isSnippetScope`. */
         isGlobalCompletion: boolean;
         isMemberCompletion: boolean;
+        /**
+         * In the absence of `CompletionEntry["replacementSpan"], the editor may choose whether to use
+         * this span or its default one. If `CompletionEntry["replacementSpan"]` is defined, that span
+         * must be used to commit that completion entry.
+         */
+        optionalReplacementSpan?: TextSpan;
```

Comment:
@andrewbranch is this correct? The examples above in the unittests show [TextSpan](https://github.com/microsoft/TypeScript/blob/main/lib/tsserverlibrary.d.ts#L7476-L7485) being used. But this is pointing to [here](https://github.com/microsoft/TypeScript/blob/main/lib/tsserverlibrary.d.ts#L3873-L3876) is it using the correct interface?

Reply:
https://github.com/microsoft/TypeScript/blob/main/src/testRunner/unittests/tsserver/metadataInResponse.ts#L83-L86 its still set to that prototype also

Reply:
Yes, internally it should be a `ts.TextSpan` and over the wire it should be a `ts.server.protocol.TextSpan`. The conversion is done on the way out in `session.ts`. It’s possible we’re missing a conversion somewhere but the types are right. What’s going on?

Reply:
> what’s going on?

im having a nightmare with a typescript server plugin not having correct completion data, mainly range data, see here https://github.com/microsoft/vscode/issues/134328

I thought maybe it needs to include This new field but TextSpan didn’t match with the examples here

Reply:
I though the culprit was here https://github.com/microsoft/typescript-styled-plugin/blob/main/src/_language-service.ts#L121-L138 because `item` has the range, but its never converted. But the types seem correct.

Looking at the types for [CompletionEntry](https://github.com/microsoft/TypeScript/blob/64b8172f0688c7567f4a880754a7f972423f092a/src/services/types.ts#L1217-L1246) and [CompletionEntryDetails](https://github.com/microsoft/TypeScript/blob/64b8172f0688c7567f4a880754a7f972423f092a/src/services/types.ts#L1248-L1259) I don't see where the range would be passed through back to VSCode

Reply:
Here’s where it gets converted on its way to VS Code https://github.com/microsoft/TypeScript/blob/9766757ee6b4df479330c170bec3a8951dd95744/src/server/session.ts#L1873

Reply:
Thanks Andrew.
I don’t suppose you know of any typescript plugin that sets this? I’ve burnt out trying to find a solution for [here](https://github.com/microsoft/typescript-styled-plugin/blob/main/src/_language-service.ts#L383). I don’t know if the offset is the virtual offset or the overall offset, should the positions be 0-indexed or match what VSCode shows?

Secondly, what does the [length](https://github.com/microsoft/TypeScript/blob/main/lib/tsserverlibrary.d.ts#L3873-L3876) argument represent? All the completions are different lengths, is it supposed to be the length of the token I’m next to?

Reply:
@andrewbranch in case you were curious, this was the fix, took me days: https://github.com/microsoft/typescript-styled-plugin/pull/156